File: examples/memory-router/openai.py
from meat_memory import MeatMemoryClient


def build_memory_context_prompt(query: str, scope_id: str, base_url: str = "http://127.0.0.1:8080") -> dict[str, str]:
    memory = MeatMemoryClient(base_url=base_url)
    context = memory.search(query=query, scope_id=scope_id, max_records=5)
    return {
        "role": "system",
        "content": f"Use the following Meat Memory context when relevant:\n{context}",
    }

File: examples/memory-router/openai.ts
import { MeatMemoryClient } from "../../sdks/typescript/src/index.ts";

export async function buildMemoryContextPrompt(input: {
  query: string;
  scopeId: string;
  baseUrl?: string;
}) {
  const memory = new MeatMemoryClient({ baseUrl: input.baseUrl });
  const context = await memory.search({
    scope_id: input.scopeId,
    query: input.query,
    max_records: 5,
  });

  return {
    role: "system",
    content: `Use the following Meat Memory context when relevant:\n${JSON.stringify(context)}`,
  };
}
